### DISCUSSION TITLE
Group footpath object selection using subtabs

### DISCUSSION THREAD

**Discussion by AaronVanGeffen** [OpenRCT2/OpenRCT2#22345] [Feature requests]
Along with the new park save format, several more object types were introduced, separating footpath objects from their railings and supports. This increased the amount of path-related object types to five, including path ornaments and banners. This is a little overbearing on the object selection tab strip.

Current situation:
![Unnamed park 2024-07-21 19-32-52](https://github.com/user-attachments/assets/b7a9b841-17c5-46ae-847f-d0aea12bda63)

I believe the situation would be improved if we group these related object types under one main path tab, along with a second set of tabs with the more specific object type. There is UI precedent for this in the ride/vehicle selection tab. While it will take a bit of refactoring, I think a situation like the one in the mockup below would vastly improve things:

Proposed situation:
![Unnamed park 2024-07-21 19-33-06](https://github.com/user-attachments/assets/916cbd2e-f7b1-4076-bc4c-a813209da822)

We could arguably do something similar for scenery objects, but I'd like to constrain this discussion to just the path objects, for now.

**Comment by leicestersquare** [comment]
Nobrainer - yes please!

The "Footpath surface" icon in the main menu is a little non descriptive though (Especially next to the walls section).

The "legacy footpath" icon would be better i'd say - or even the "Build footpath" icon from the main game.

**Reply by AaronVanGeffen** [reply]
Agreed, it would be nice to have a different icon for the 'main' tab group. The mockup doesn't include that (yet) as I'd just moved the existing widgets around in Photoshop 😄

**Comment by 733737** [comment]
I think this is a good idea

**Comment by Gymnasiast** [comment]
I fear this may be confusing, as it makes discovery harder.

I have a different proposal: allow for some space inbetween tabs and group similar tabs that way. That would also help for scenery and land/water.

**Reply by AaronVanGeffen** [reply]
> I fear this may be confusing, as it makes discovery harder.

Hmm, I don't share this fear. It's not like we're grouping random things together, right? I believe path-related options can still easily be found this way. Especially if we group other object types logically as well, all it takes is a click through the main ones, in the absolute worse case.

> I have a different proposal: allow for some space inbetween tabs and group similar tabs that way. That would also help for scenery and land/water.

While that might solve the immediate problem, it doesn't solve the main issue with horizontal space being scarce enough as it is — it would actually need more.

Especially with us adding at least two new object types in the near future, and probably more, I would like to move towards something that is viable long term, even if it might take some getting used to in the short term. (I really think it will be fine, though.)

**Reply by Gymnasiast** [reply]
I'm not sure I see stuff the same as you, but if we're going on this path, I would suggest doing the same for scenery and land/water etc. at the same time. Because that would at least keep things consistent.

**Reply by AaronVanGeffen** [reply]
> if we're going on this path, I would suggest doing the same for scenery and land/water etc. at the same time.

I agree, and wrote as much in the last paragraph of the OP. To keep the discussion relatively concise, I thought it best to limit it to the path-related objects, for now, though.

**Comment by Basssiiie** [comment]
I think it's a good idea. It makes the top tab a bit less cluttered. I think it also makes it a bit more obvious that split paths and legacy paths are in different tabs, when looking for a specific path object. 🙂

**Comment by AaronVanGeffen** [comment]
This now has a working implementation in #22352.
